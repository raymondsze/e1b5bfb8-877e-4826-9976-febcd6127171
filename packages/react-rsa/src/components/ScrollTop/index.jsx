import PropTypes from 'prop-types';
import React from 'react';
import { withRouter } from 'react-router';

// Alternative solution of react-router-scroll for react-router v4
class ScrollTop extends React.Component {
  static propTypes = {
    location: PropTypes.shape({}).isRequired,
    children: PropTypes.element.isRequired,
  };

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      // scroll to top if location is different
      window.scrollTo(0, 0);
    }
  }

  render() {
    const { children } = this.props;
    return React.Children.only(children);
  }
}

export default withRouter(ScrollTop);
