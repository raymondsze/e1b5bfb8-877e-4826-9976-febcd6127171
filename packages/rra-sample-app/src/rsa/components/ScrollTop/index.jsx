import React from 'react';
import { withRouter } from 'react-router';
import { Route } from 'react-router-dom';

// Alternative solution of react-router-scroll for react-router v4
class ScrollTop extends React.Component {
  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      // scroll to top if location is different
      window.scrollTo(0, 0)
    }
  }

  render() {
    const { children, location } = this.props;
    return React.Children.only(children);
  }
}

export default withRouter(ScrollTop);
