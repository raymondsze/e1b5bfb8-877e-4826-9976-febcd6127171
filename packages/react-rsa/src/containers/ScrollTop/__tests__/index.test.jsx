import React from 'react';
import { MemoryRouter } from 'react-router';
import { mount, shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { fromJS } from 'immutable';
import ScrollTop from '../index';

it('ScrollTop should only render children', () => {
  const location = {
    pathname: 'location',
  };
  const wrapper = mount(
    <ScrollTop location={location}>
      <div>Test Scroll</div>
    </ScrollTop>
  );
  // this should not trigger render
  wrapper.setProps({ location });
  wrapper.setProps({
    location: {
      pathname: 'new-location',
    },
  });
});
