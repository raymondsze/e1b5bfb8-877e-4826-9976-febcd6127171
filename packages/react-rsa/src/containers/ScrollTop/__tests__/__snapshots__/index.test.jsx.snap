// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollTop should only render children 1`] = `null`;

exports[`ScrollTop should only render children 2`] = `
<ScrollTop
  location={
    Object {
      "pathname": "location",
    }
  }
>
  <div>
    Test Scroll
  </div>
</ScrollTop>
`;
